### ISSUE TITLE
Test with FreeMocap data

### ISSUE THREAD

**Issue by roaldarbol** [roaldarbol/tidymocap#6]
Use the data structure from https://github.com/freemocap/freemocap/wiki with their `openPoseSkel_3D`.

**Comment by jonmatthis** [comment]
Hey neat! 

If you can orient us towards a good place to point our data, I'm sure it wouldn't be hard for us to write a simple wrapper that re-shapes our data to fit your pipelines!

Is there a particular entry point to your software that you had in mind? Something that reads in `3d data about skeleton joint locations` or something like that? 

What is the expected structure of the `anipose_raw` data that you reference in the `Read_Me`?

**Comment by roaldarbol** [comment]
Hey @jonmatthis, thanks for reaching out! 
I'll need to work a bit on my documentation, but I think this is a good occasion. ;-)

Regarding entry point (if I understand the question right), yes, I guess so. The idea is to have a few different `tidy` functions, so people are able to import whatever mocap data - also marker-based. They are basically just a way to streamline up whatever weird output is generated by different pieces of software. The raw structure expected from `anipose` is currently less than convenient as some of the column identifiers are actually inside the data frame: <img width="809" alt="Screenshot 2022-06-07 at 09 48 50" src="https://user-images.githubusercontent.com/25629697/172325843-c29654c1-d4a0-4a0f-85fc-52a92754da22.png">

The tidy format is a so-called long format, and consists of the columns `frame`(/`time`), `bodypart`, `x`, `y`, `z` (if present), `likelihood`.
<img width="433" alt="Screenshot 2022-06-07 at 09 52 32" src="https://user-images.githubusercontent.com/25629697/172326537-3df64e2f-9a77-4972-af49-0f8ec6f55f46.png">
 
I still haven't been able to work with an actual 3D data frame, so you'd be able to export a pandas to csv I'd really appreciate it!

Additionally, the idea is to have a "library" of skeletons, e.g. MediaPipe, OpenPose, Qualisys, Vicon, etc. as YAML/TOML files, with the ability to manually enter a different skeleton (in case people, like myself, come with DeepLabCut output), mostly for plotting purposes.

**Comment by jonmatthis** [comment]
Yes, sounds good! We can definitely write some functions to transform and output data from `freemocap/mediapipe` into this format. 

We've also been starting to work on a pipeline that will take skeletons from disparate sources and transforming them into a generic skeleton format, roughly following Winter's anthropometry skeleton: 
![image](https://user-images.githubusercontent.com/15314521/172639405-fff28f5d-351c-4d6d-b769-0ef0016c8fb2.png)

Properly executed, that should allow for easier comparison of data from different tracking methods (mediapipe, openpose, Vicon, etc).

**Comment by roaldarbol** [comment]
Sounds great!

Please do keep me in the loop on your progress on that pipeline. Would be super useful, and I could write an R implementation if you manage in Python.
